python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import json
+import logging
+from typing import Dict, List, Optional, Union
+
+import aiohttp
+from pydantic import HttpUrl
+
+from semantic_kernel.connectors.ai.ai_request_settings import AIRequestSettings
+from semantic_kernel.connectors.ai.ai_service_client_base import AIServiceClientBase
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.connectors.ai.ollama.ollama_request_settings import OllamaChatRequestSettings
+from semantic_kernel.connectors.ai.text_completion_client_base import (
+    TextCompletionClientBase,
+)
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class OllamaChatCompletion(TextCompletionClientBase, ChatCompletionClientBase, AIServiceClientBase):
+    """
+    Initializes a new instance of the OllamaChatCompletion class.
+
+    Make sure to have the ollama service running either locally or remotely.
+
+    Arguments:
+        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/chat
+    """
+
+    url: HttpUrl = "http://localhost:11434/api/chat"
+    session: Optional[aiohttp.ClientSession] = None
+
+    async def complete_chat_async(
+        self,
+        messages: List[Dict[str, str]],
+        request_settings: OllamaChatRequestSettings,
+        **kwargs,
+    ) -> Union[str, List[str]]:
+        request_settings.messages = messages
+        request_settings.stream = False
+        if self.session:
+            async with self.session.post(str(self.url), json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                response_object = await response.json()
+                return response_object.get("message", {"content": None}).get("content", None)
+        async with aiohttp.ClientSession() as session:
+            async with session.post(str(self.url), json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                response_object = await response.json()
+                return response_object.get("message", {"content": None}).get("content", None)
```

Comment:
Seeing that this `if self.session...with self.session...else...with create session` is used multiple times and the blocks inside them is the same. Would it make sense to create a context manager that takes care of that so the statements are not duplicated?

Something like this:
```python

class AsyncSession:
    def __init__(self, session: Optional[aiohttp.ClientSession] = None):
		self.session = session if session else aiohttp.ClientSession()
		
	async def __aenter__(self):
		return self.session
	
	async def __aexit__(self, exc_type, exc_value, exc_tb):
		await self.session.close()
```

then this block can be reduced to:
```python
async with AsyncSession(self.session) as session:
	async with session.post(str(self.url), json=request_settings.prepare_settings_dict()) as response:
		response.raise_for_status()
		response_object = await response.json()
		return response_object.get("message", {"content": None}).get("content", None)
```

Reply:
I remember seeing this changes and now they are gone and the duplicated logic is back in.